lib/inspec/resources/zfs_dataset.rb

```diff
@@ -16,16 +16,20 @@ class ZfsDataset < Inspec.resource(1)
     EXAMPLE
 
     def initialize(zfs_dataset)
-      return skip_resource "The `zfs_dataset` resource is not supported on your OS yet." unless inspec.os.bsd?
+      return skip_resource "The `zfs_dataset` resource is not supported on your OS yet." unless inspec.os.bsd? || inspec.os.linux?
```

Comment:
@kannanr How about just running the `zfs list dataset` or something to see if it is supported?

Reply:
@ptyagi16 For now, the code covers both bsd and linux based systems. and for specific platforms [Please check the supported OS here](https://docs.chef.io/inspec/resources/os/#osfamily-helpers).

Note: As part of your another comment, I've updated the document as well with the link.